ports/stm32/usbd_cdc_interface.c

```diff
@@ -236,6 +236,25 @@ void HAL_PCD_SOFCallback(PCD_HandleTypeDef *hpcd) {
     }
 }
 
+bool usbd_cdc_rx_buffer_any(usbd_cdc_itf_t *cdc) {
+    int get = cdc->rx_buf_get, put = cdc->rx_buf_put;
+    int remaining = (get - put) + (-((int) (get <= put)) & USBD_CDC_RX_DATA_SIZE);
+    return remaining > (CDC_DATA_MAX_PACKET_SIZE*2);
```

Comment:
Why did you use this formula, with the `*2`?  Does it need that much space?  I tried it without the `*2` and it seems to work fine (tested on F733 in HS mode).

With USB HS, `CDC_DATA_MAX_PACKET_SIZE` is 512 and total buffer size `USBD_CDC_RX_DATA_SIZE` is 1024, so this test with `*2` always fails.

Reply:
Yep I just commented a few minutes ago above regarding this. I've just re-tested myself and getting good results, thinking of leaving it as 
```
bool usbd_cdc_rx_buffer_full(usbd_cdc_itf_t *cdc) {
    int get = cdc->rx_buf_get, put = cdc->rx_buf_put;
    int remaining = (get - put) + (-((int) (get <= put)) & USBD_CDC_RX_DATA_SIZE);
    return remaining < (CDC_DATA_MAX_PACKET_SIZE-1);
}
```
Just an extra -1 in there to give a bit of extra head room.

Reply:
The fact I couldn't reproduce the overflow failure with a -1 make me much more confident I fixed whatever caused my earlier issues...